Gems/Atom/RHI/Code/Source/RHI/CpuProfilerImpl.cpp

```diff
@@ -141,12 +144,47 @@ namespace AZ
             return m_timeRegionMap;
         }
 
+        bool CpuProfilerImpl::BeginContinuousCapture()
+        {
+            if (m_continuousCaptureInProgress)
```

Comment:
After loading the value in this atomic, nothing stops a different thread from entering the same place and setting `m_continuousCaptureInProgress` to true on line 156, before _this thread_ gets to the same line. While this is technically benign because the atomic will stay true in both cases, you might have two threads competing to deactivate the capture later because they both believe they started the capture.

The pattern to employ here is the atomic exchange, where you do something like

```cpp
if (m_continuousCaptureInProgress.compare_exchange_strong(false, true))
{
    // We atomically transitioned the value of m_continuousCaptureInProgress from false to true, so are responsible for the capture and can start it
}
else
{
    return false;
}
```